libs/features/mas/src/lana.js

```diff
@@ -61,26 +64,36 @@ const lanaAppender = {
         });
 
         if (errors.length) {
-            payload += ' ';
-            payload += errors.map(serializeValue).join(' ');
+            payload += ' ' + errors.map(serializeValue).join(' ');
         }
 
-        // TODO: Lana backend extracts referer from the header sent,
-        // sometimes it includes page path and query, sometimes - not
         const { pathname, search } = window.location;
-        payload += `${delimiter}page=`;
-        payload += pathname + search;
+        payload += `${config.delimiter}page=${pathname}${search}`;
 
         if (values.length) {
-            payload += `${delimiter}facts=`;
+            payload += `${config.delimiter}facts=`;
             payload += JSON.stringify(values, serializeParam);
         }
 
         if (!seenPayloads.has(payload)) {
             seenPayloads.add(payload);
-            window.lana?.log(payload, { sampleRate, tags, clientId });
+            window.lana?.log(payload, config);
         }
     },
 };
 
-export { Defaults, lanaAppender };
+// Allow dynamic config updates
+function updateConfig(newConfig) {
+    Object.assign(
+        config,
+        Object.fromEntries(
+            Object.entries(newConfig).filter(
+                ([key, value]) =>
+                    key in config &&
+                    !['', NaN, null, undefined].includes(value), // 0 is a valid value.
```

Comment:
`NaN` values will not be filtered out here since they're not equal to any value, including itself. Instead we can use `Number.isNaN(value)` to correctly check for NaN.
```suggestion
                    function updateConfig(newConfig) {
    Object.assign(
        config,
        Object.fromEntries(
            Object.entries(newConfig).filter(
                ([key, value]) =>
                    key in config &&
                    value !== '' &&
                    value !== null &&
                    value !== undefined &&
                    !Number.isNaN(value), // Correctly exclude NaN
            ),
        ),
    );
}
```

Reply:
`[NaN].includes(NaN) = true` though. Still to be on the safe side, I prefer your version.